### PR TITLE
feat: Static CSS Optimization - Block Based CSS loading.

### PR DESCRIPTION
**Pull request by RajkiranBagal**
### Description
 - Trello task - https://trello.com/c/AazLVssY/265-static-css-optimization-block-specific-loading

### Types of changes
New feature (non-breaking change which adds functionality)

### How has this been tested?
 - Added in Trello task.

### Checklist:
- [x] My code is tested
- [x] I've included any necessary tests 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-block-helper.php [hunk 1]
```diff
@@ -5491,7 +5701,37 @@ public static function get_lottie_css( $attr, $id ) {
...
+
+			$combined_selectors = array(
+				'desktop' => array(),
+				'tablet'  => array(),
+				'mobile'  => array(),
+			);
+
+			$google_map_static_css = self::get_block_static_css( 'google-map' );
+
+			$css = UAGB_Helper::generate_all_css( $combined_selectors, '.uagb-block-' . $id );
```
Comment (vrundakansara) [inline classes/class-uagb-block-helper.php:5730]: Remove this line of code if there is no dynamic CSS associated with it.

File: dist/blocks.style.css [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
+.wp-block-uagb-advanced-heading{padding:0;margin:0 auto}.wp-block-uagb-advanced-heading .uagb-heading-text{margin:0;text-align:center}.wp-block-uagb-advanced-heading .uagb-separator-wrap{font-size:0;text-align:center}.wp-block-uagb-advanced-heading .uagb-separator{border-top-style:solid;display:inline-block;border-top-width:2px;width:5%;margin:0px 0px 10px 0px}.wp-block-uagb-advanced-heading .uagb-desc-text{margin:0;text-align:center}
```
Comment (vrundakansara) [inline dist/blocks.style.css:2]: If this file is not required let's delete it entirely.

File: classes/class-uagb-block-helper.php [hunk 3]
```diff
@@ -4563,7 +4735,13 @@ public static function get_marketing_btn_css( $attr, $id ) {
 
 			$base_selector = ( $attr['classMigrate'] ) ? '.uagb-block-' : '#uagb-marketing-btn-';
 
-			return UAGB_Helper::generate_all_css( $combined_selectors, $base_selector . $id );
+			$marketing_button_static_css = self::get_block_static_css( 'marketing-button' );
+
+			$css = UAGB_Helper::generate_all_css( $combined_selectors, $base_selector . $id );
+
+			$css['desktop'] = $marketing_button_static_css . $css['desktop'];
```
Comment (vrundakansara) [inline classes/class-uagb-block-helper.php:4742]: I see we have added all the static CSS to the `desktop` index. Do check if any media-query-based CSS is running or not that is added in the static CSS.

**Review by vrundakansara — CHANGES_REQUESTED** [review]

File: dist/blocks.style.css (hunk 2 above)
Comment (RajkiranBagal) [reply dist/blocks.style.css:2]: Loaded the file in the Editor. As it is needed there.

File: classes/class-uagb-block-helper.php (hunk 3 above)
Comment (RajkiranBagal) [reply classes/class-uagb-block-helper.php:4742]: Media Query Based CSS will automatically work as we load the Desktop CSS as it is without adding any Media query for Desktop.

File: classes/class-uagb-block-helper.php (hunk 1 above)
Comment (RajkiranBagal) [reply classes/class-uagb-block-helper.php]: Done

**Pull request merged**